### PR TITLE
Add support for PHP 8.2, drop support for PHP 7.4

### PR DESCRIPTION
**Pull request by ghostwriter**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

- Update composer config `platform.php` to `8.0.99`
- Update composer `PHP` to `~8.0.0 || ~8.1.0 || ~8.2.0`
- Ignore PHP platform requirements via `.laminas-ci.json`
- Update code to PHP8

### CODE REVIEW

File: .laminas-ci.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
         "--no-plugins"
     ],
     "ignore_php_platform_requirements": {
-        "8.0": false,
-        "8.1": false
+        "8.2": true
```
Comment (Ocramius) [inline .laminas-ci.json]: We shouldn't have this on the skeleton: this means trouble for people installing the skeleton :thinking:

**Review by Ocramius — APPROVED** [review]
LGTM besides the suppression during installation: can we try without that one? :thinking:

File: .laminas-ci.json (hunk 1 above)
Comment (ghostwriter) [reply .laminas-ci.json]: We want to ensure dependencies are installed correctly and that all QA tests/tools can run in the CI pipeline.

We should instead add a composer script that runs on `post-create-project-cmd` instead, which will delete both the `.laminas-ci` directory and `.laminas-ci.json` file.

``` JSON
"post-create-project-cmd": [
    "php bin/remove-laminas-ci-artifacts.php",
],
```

Thoughts?

Comment (Ocramius) [reply .laminas-ci.json]: Hmm, if 8.2 does not apply cleanly, then perhaps yes.

My question here was rather whether CI would work without this specific `true`, and whether we can instead do a couple more releases to get there instead.

Comment (Ocramius) [reply .laminas-ci.json]: Ok, so indeed it seems like mezzio-tooling is the last missing piece

Comment (ghostwriter) [reply .laminas-ci.json]: Bumped mezzio/mezzio-tooling to ^2.8

**Review by Ocramius — APPROVED** [review]
LGTM, thanks @ghostwriter!

**Pull request merged**